netdeploy/networkTemplate.go

```diff
@@ -221,17 +223,49 @@ func (t NetworkTemplate) Validate() error {
 		}
 	}
 
+	// At least one relay is required
 	if len(t.Nodes) > 1 && countRelayNodes(t.Nodes) == 0 {
 		return fmt.Errorf("invalid template: at least one relay is required when more than a single node presents")
 	}
 
+	// Validate JSONOverride decoding
+	for _, cfg := range t.Nodes {
+		local := config.GetDefaultLocal()
+		err := decodeJSONOverride(cfg.ConfigJSONOverride, &local)
+		if err != nil {
+			return fmt.Errorf("invalid template: unable to decode JSONOverride: %w", err)
+		}
+	}
+
+	// Follow nodes cannot be relays
+	for _, cfg := range t.Nodes {
+		if cfg.IsRelay && isEnableFollowMode(cfg.ConfigJSONOverride) {
+			return fmt.Errorf("invalid template: follow nodes may not be relays")
+		}
+	}
+
 	if t.Genesis.DevMode && len(t.Nodes) != 1 {
-		return fmt.Errorf("invalid template: DevMode should only have a single node")
+		if countRelayNodes(t.Nodes) != 1 {
```

Comment:
why do we need to run a relay node for dev mode?

Reply:
If there's a follower node, it needs to connect to the relay.